pkg/daemon/update.go

```diff
@@ -217,6 +219,56 @@ func addExtensionsRepo(osImageContentDir string) error {
 	return nil
 }
 
+// podmanRemove kills and removes a container
+func podmanRemove(cid string) {
+	// Ignore errors here
+	exec.Command("podman", "kill", cid).Run()
+	exec.Command("podman", "rm", "-f", cid).Run()
+}
+
+func podmanCopy(imgURL, osImageContentDir string) (err error) {
+	// make sure that osImageContentDir doesn't exist
+	os.RemoveAll(osImageContentDir)
+
+	// Pull the container image
+	var authArgs []string
+	if _, err := os.Stat(kubeletAuthFile); err == nil {
+		authArgs = append(authArgs, "--authfile", kubeletAuthFile)
+	}
+	args := []string{"pull", "-q"}
+	args = append(args, authArgs...)
+	args = append(args, imgURL)
+	_, err = pivotutils.RunExt(numRetriesNetCommands, "podman", args...)
+	if err != nil {
+		return
+	}
+
+	// create a container
+	var cidBuf []byte
+	containerName := pivottypes.PivotNamePrefix + string(uuid.NewUUID())
+	cidBuf, err = runGetOut("podman", "create", "--net=none", "--annotation=org.openshift.machineconfigoperator.pivot=true", "--name", containerName, imgURL)
+	if err != nil {
+		return
+	}
+
+	// only delete created container, we will delete container image later as we may need it for podmanInspect()
+	defer podmanRemove(containerName)
+
+	// copy the content from create container locally into a temp directory under /run/machine-os-content/
+	cid := strings.TrimSpace(string(cidBuf))
+	args = []string{"cp", fmt.Sprintf("%s:/", cid), osImageContentDir}
+	_, err = pivotutils.RunExt(numRetriesNetCommands, "podman", args...)
+
+	// Set selinux context to var_run_t to avoid selinux denial
```

Comment:
One random aside: SELinux is really good about protecting the host from containers, and avoiding containers interacting with other containers that they shouldn't.

But, we're trying to cross these two worlds, and that's the source of pain.

Reply:
yeah! we have come long way forward with running everything in container context. With always improvement in these awesome software, we can hope to reduce the pain in future.